src/v/security/jwt.h

```diff
@@ -458,6 +480,144 @@ inline result<verifier> make_rs256_verifier(
     return verifier{rs256_verifier{key}};
 }
 
+using verifiers = absl::flat_hash_map<
+  ss::sstring,
+  detail::verifier,
+  detail::string_viewable_hasher,
+  detail::string_viewable_compare>;
+
+inline result<verifiers>
+make_verifiers(jwks const& jwks, CryptoPP::AutoSeededRandomPool& rng) {
+    auto keys = jwks.keys();
+    verifiers vs;
+    for (auto const& key : keys) {
+        // Alg is required
+        auto alg = detail::string_view(key, "alg");
+        if (!alg) {
+            return errc::jwk_invalid;
+        }
+
+        // Use is optional, but must be sig if it exists
+        auto use = detail::string_view(key, "use");
+        if (use.value_or("sig") != "sig") {
+            continue;
+        }
+
+        using factory = result<verifier> (*)(
+          json::Value const&, CryptoPP::AutoSeededRandomPool&);
+        auto v = string_switch<std::optional<factory>>(*alg)
+                   .match(rs256_str, &make_rs256_verifier)
+                   .default_match(std::optional<factory>{});
```

Comment:
😄

Reply:
> 😄 

It's somewhat over-engineered for a single verifier, but I also have an implementation for ES256, just not tests!

Reply:
Pretty satisfying nonetheless 👍